src/pset/map/input.rs

```diff
@@ -258,6 +260,95 @@ pub struct Input {
     pub unknown: BTreeMap<raw::Key, Vec<u8>>,
 }
 
+/// A Signature hash type for the corresponding input. As of taproot upgrade, the signature hash
+/// type can be either [`SigHashType`] or [`SchnorrSigHashType`] but it is not possible to know
+/// directly which signature hash type the user is dealing with. Therefore, the user is responsible
+/// for converting to/from [`PsbtSighashType`] from/to the desired signature hash type they need.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```

Comment:
This derived will be a struct with an inner field that is a number, I doubt that is what we'd want. I think the other sighash types either use the `Display`/`FromStr` to serde-serialize as string, or just serialize as the number.

Reply:
Indeed, I will add this in a followup commit to this PR